### PR TITLE
FNOModel with Meshgrids and Normalization

### PR DESCRIPTION
**Pull request by spellsharp**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by JoseAntonioSiguenza** [comment]
Can you please fix failing unit-test? `deepchem/models/torch_models/fno.py::deepchem.models.torch_models.fno.FNOModel FAILED [ 26%].`

I will give a complete review once CI torch tests are completed.

File: deepchem/models/torch_models/fno.py [hunk 1]
```diff
@@ -83,3 +84,243 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
...
+            torch.linspace(0, 1, steps=s, device=x.device) for s in spatial
+        ]
+        mesh = torch.meshgrid(*coords, indexing='ij')
+        grid = torch.stack(mesh, dim=0)  # shape (dims, *spatial)
+        # Repeat for batch dimension and add spatial dimensions
+        grid = grid.unsqueeze(0).expand(batch_size, -1, *spatial)
+        return grid
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if x.ndim != self.dims + 2:
```
Comment (JoseAntonioSiguenza) [inline deepchem/models/torch_models/fno.py]: Please add docstrings for `forward`method

**Comment by JoseAntonioSiguenza** [comment]
Minor comment on dosctrings. In general, the code looks fine, but some `unit-tests` in CI got cancelled due to memory issues. LGTM under green CI.

File: deepchem/models/torch_models/fno.py [hunk 2]
```diff
@@ -83,3 +84,255 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
...
+
+
+class FNO(nn.Module):
+    """Base implementation of Fourier Neural Operator, inheriting from the Torch nn.Module class.
+
+    Fourier Neural Operator (FNO) is a neural network architecture for learning
+    mappings between function spaces. It uses spectral convolutions in Fourier
+    space to capture global dependencies efficiently, making it particularly
+    effective for solving partial differential equations (PDEs).
+
```
Comment (rbharath) [inline deepchem/models/torch_models/fno.py]: Can be done in later PR, but I would like more documentation about the meshgrid. Remeber this is the top level class people will see
